### PR TITLE
op-geth builder clean up

### PR DESCRIPTION
**Pull request by avalonche**

### CODE REVIEW

File: builder/builder.go [hunk 1]
```diff
@@ -0,0 +1,606 @@
...
+		payload, err := b.eth.BuildBlock(attrs)
+		if err != nil {
+			log.Warn("Failed to build block", "err", err)
+			return
+		}
+
+		sealedAt := time.Now()
+		queueMu.Lock()
+		defer queueMu.Unlock()
+		if payload.ExecutionPayload.BlockHash != queueLastSubmittedHash && payload.BlockValue.Cmp(queueBestBlockValue) > 0 {
```
Comment (jinmel) [inline builder/builder.go]: if the block is empty, the `payload.BlockValue` is 0. 
payload.BlockValue.Cmp(queueBestBlockValue) > 0 should be payload.BlockValue.Cmp(queueBestBlockValue) >= 0 to account for empty blocks

File: builder/eth_service.go [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+		log.Error("Failed to build payload", "err", err)
+		return nil, err
+	}
+
+	resCh := make(chan *engine.ExecutionPayloadEnvelope, 1)
+	go func() {
+		resCh <- payload.ResolveFull()
+	}()
+
+	timer := time.NewTimer(2 * time.Second)
```
Comment (jinmel) [inline builder/eth_service.go]: I see that it's set to 4 seconds for L1 builder. should it be dependent on block time? if block building takes longer than 2 seconds it's already a failure

File: builder/builder.go (hunk 1 above)
Comment (jinmel) [reply builder/builder.go]: I fixed it to `payload.BlockValue.Cmp(queueBestBlockValue) >= 0`

**Review by liamaharon — COMMENTED** [review]
Looking really clean!

Can we please also include the builder metrics present in the `flashbots-builder` branch? e.g. https://github.com/flashbots/op-geth/blob/flashbots-builder/miner/metrics.go https://github.com/flashbots/op-geth/blob/ab3a6b87ff107819cb166667ec99252ae8cc2d9a/metrics/metrics.go#L32

File: go.mod [hunk 3]
```diff
@@ -27,15 +29,17 @@ require (
 	github.com/ethereum/c-kzg-4844 v1.0.0
 	github.com/ethereum/go-verkle v0.1.1-0.20240306133620-7d920df305f0
 	github.com/fatih/color v1.16.0
-	github.com/ferranbt/fastssz v0.1.2
+	github.com/ferranbt/fastssz v0.1.3
```
Comment (ferranbt) [inline go.mod:32]: :)

File: cmd/utils/flags.go [hunk 4]
```diff
@@ -669,6 +670,62 @@ var (
...
+	}
+	BuilderBeaconEndpoints = &cli.StringFlag{
+		Name:     "builder.beacon_endpoints",
+		Usage:    "Comma separated list of beacon endpoints to connect to for beacon chain data",
+		EnvVars:  []string{"BUILDER_BEACON_ENDPOINTS"},
+		Value:    "http://127.0.0.1:5052",
+		Category: flags.BuilderCategory,
+	}
+	BuilderBlockTime = &cli.DurationFlag{
+		Name:     "builder.block_time",
```
Comment (ferranbt) [inline cmd/utils/flags.go:715]: I would prefer if we can query this from the op-node.

File: cmd/utils/flags.go [hunk 5]
```diff
@@ -669,6 +670,62 @@ var (
...
+	BuilderGenesisForkVersion = &cli.StringFlag{
+		Name:     "builder.genesis_fork_version",
+		Usage:    "Gensis fork version.",
+		EnvVars:  []string{"BUILDER_GENESIS_FORK_VERSION"},
+		Value:    "0x00000000",
+		Category: flags.BuilderCategory,
+	}
+	BuilderBeaconEndpoints = &cli.StringFlag{
+		Name:     "builder.beacon_endpoints",
+		Usage:    "Comma separated list of beacon endpoints to connect to for beacon chain data",
```
Comment (ferranbt) [inline cmd/utils/flags.go:709]: Is it beacon endpoint or op-node endpoint?

File: cmd/utils/flags.go [hunk 6]
```diff
@@ -669,6 +670,62 @@ var (
...
+		Category: flags.BuilderCategory,
+	}
+	BuilderListenAddr = &cli.StringFlag{
+		Name:     "builder.listen_addr",
+		Usage:    "Listening address for builder endpoint",
+		EnvVars:  []string{"BUILDER_LISTEN_ADDR"},
+		Value:    ":28545",
+		Category: flags.BuilderCategory,
+	}
+	BuilderGenesisForkVersion = &cli.StringFlag{
```
Comment (ferranbt) [inline cmd/utils/flags.go:700]: Is this required by op-geth? Also, if you have the address of the beacon node you can query this field.

**Review by ferranbt — APPROVED** [review]

File: cmd/utils/flags.go (hunk 5 above)
Comment (avalonche) [reply cmd/utils/flags.go:709]: its actually the payload attribute endpoint, which doesn't have BuilderGenesisForkVersion and BlockTime information

File: builder/eth_service.go (hunk 2 above)
Comment (avalonche) [reply builder/eth_service.go]: yeah using block time as default for now, can update as additional param later

**Pull request merged**